File: apps/go/fizzbuzz/fizzbuzz.go
// Package fizzbuzz provides FizzBuzz generation and output helpers.
package fizzbuzz

import (
	"fmt"
	"io"

	"github.com/k2works/getting-started-tdd/apps/go/application"
	"github.com/k2works/getting-started-tdd/apps/go/domain/model"
	type_ "github.com/k2works/getting-started-tdd/apps/go/domain/type_"
)

// Generate は FizzBuzz の文字列を返します。
func Generate(number int) string {
	return type_.FizzBuzzType01{}.Generate(number).String()
}

// GenerateList は start から end までの FizzBuzz 結果をスライスで返します。
func GenerateList(start, end int) []string {
	count := end - start + 1
	command := application.NewFizzBuzzListCommand(type_.FizzBuzzType01{}, count)
	return command.Execute().(*model.FizzBuzzList).ToStringSlice()
}

// Print は FizzBuzz の結果を writer に出力します。
func Print(w io.Writer) error {
	for _, s := range GenerateList(1, 100) {
		if _, err := fmt.Fprintln(w, s); err != nil {
			return err
		}
	}
	return nil
}

File: apps/typescript/src/fizzbuzz/fizz-buzz.ts
import { FizzBuzzListCommand } from "./application/fizz-buzz-list-command.js";
import { FizzBuzzValueCommand } from "./application/fizz-buzz-value-command.js";
import { FizzBuzzList } from "./domain/model/fizz-buzz-list.js";
import { FizzBuzzValue } from "./domain/model/fizz-buzz-value.js";
import { FizzBuzzType } from "./domain/type/fizz-buzz-type.js";
import { FizzBuzzType01 } from "./domain/type/fizz-buzz-type-01.js";

export class FizzBuzz {
  private readonly _type: FizzBuzzType;

  constructor(type?: FizzBuzzType) {
    this._type = type ?? new FizzBuzzType01();
  }

  get type(): FizzBuzzType {
    return this._type;
  }

  generate(number: number): FizzBuzzValue {
    return new FizzBuzzValueCommand(this._type, number).execute();
  }

  generateList(count: number): FizzBuzzList {
    return new FizzBuzzListCommand(this._type, count).execute();
  }

  printFizzBuzz(count: number): void {
    for (const value of this.generateList(count)) {
      console.log(value.toString());
    }
  }
}
